policy/lib/tekton/task_test.rego

```diff
@@ -469,3 +484,118 @@ slsav1_attestation_local_spec := {
 }
 
 _bundle := "registry.img/spam@sha256:4e388ab32b10dc8dbc7e28144f552830adc74787c1e2c0824032078a79f227fb"
+
+slsav1_task(name) := task if {
+	parts := regex.split(`[\[\]=]`, name)
+	not parts[1]
+	pipeline_task_name := sprintf("%s-p", [name])
+	unnamed_task := {
+		"metadata": {"name": pipeline_task_name},
+		"spec": slsav1_attestation_local_spec,
+		"status": {"conditions": [{
+			"type": "Succeeded",
+			"status": "True",
+		}]},
+	}
+	task := json.patch(unnamed_task, [{
+		"op": "replace",
+		"path": "/spec/taskRef/name",
+		"value": name,
+	}])
+}
+
+slsav1_task(name) := task if {
+	parts := regex.split(`[\[\]=]`, name)
+	parts[1]
+	task_name := parts[0]
+	pipeline_task_name := sprintf("%s-p", [task_name])
+	unnamed_task := {
+		"metadata": {"name": pipeline_task_name},
+		"spec": slsav1_attestation_local_spec,
+		"status": {"conditions": [{
+			"type": "Succeeded",
+			"status": "True",
+		}]},
+	}
+	task := json.patch(unnamed_task, [
+		{
+			"op": "replace",
+			"path": "/spec/taskRef/name",
+			"value": task_name,
+		},
+		{
+			"op": "replace",
+			"path": "/spec/params",
+			"value": [{"name": parts[1], "value": parts[2]}],
+		},
+	])
+}
+
+# create a task and add a bundle to it
+slsav1_task_bundle(name, bundle) := task if {
+	not name.spec
+	task := json.patch(slsav1_task(name), [{
+		"op": "add",
+		"path": "/spec/taskRef/bundle",
+		"value": bundle,
+	}])
+}
+
+# add a bundle to an existing task
+slsav1_task_bundle(name, bundle) := task if {
+	name.spec
+	task := json.patch(name, [{
+		"op": "add",
+		"path": "/spec/taskRef/bundle",
+		"value": bundle,
+	}])
+}
+
+slsav1_task_steps(name, steps) := json.patch(
+	slsav1_task(name),
+	[
+		{
+			"op": "add",
+			"path": "/status/taskSpec",
+			"value": {},
+		},
+		{
+			"op": "add",
+			"path": "/status/taskSpec/steps",
+			"value": steps,
+		},
+	],
+)
+
+# results are an array of dictionaries with keys, "name", "type", "value"
+slsav1_task_result(name, results) := json.patch(
+	slsav1_task(name),
+	[{
+		"op": "add",
+		"path": "/status/taskResults",
+		"value": results,
+	}],
+)
+
+# results are an array of dictionaries with keys, "name", "type", "value"
+slsav1_task_result_ref(name, results) := json.patch(
+	slsav1_task(name),
+	[{
+		"op": "add",
+		"path": "/status/taskResults",
+		"value": _marshal_slsav1_results(results),
+	}],
+)
+
+_marshal_slsav1_results(results) := [r |
+	some result in results
+	r := {"name": result.name, "type": result.type, "value": json.marshal(result.value)}
+]
+
+resolved_dependencies(tasks) := [r |
+	some task in tasks
+	r := {
+		"name": "pipelineTask",
+		"content": json.marshal(task),
```

Comment:
We need to base64 encode this, right?

Reply:
yes, good catch. that's a pretty big mistake. I'll fix.